api/graphite.go

```diff
@@ -691,10 +701,17 @@ func (s *Server) executePlan(ctx context.Context, orgId uint32, plan expr.Plan)
 	for k := range data {
 		sort.Sort(models.SeriesByTarget(data[k]))
 	}
+	stats["preRunLogicTime"] = util.MsSince(preRunLogic)
 
 	preRun := time.Now()
 	out, err = plan.Run(data)
 	planRunDuration.Value(time.Since(preRun))
+	stats["planRunTime"] = util.MsSince(preRun)
+	stats["executePlanTime"] = util.MsSince(preExecutePlan)
+	for i,_ := range out {
+		out[i].Stats = stats
+	}
```

Comment:
this is a sign that perhaps the Series is not the right place to put the stats.
i just suggest to define them in a dedicated struct type and keep 1 instance per request